### PR TITLE
[microNPU] Add support for MEAN with uint8 ifm

### PR DESCRIPTION
**Pull request by ilyag-grovety**
This PR involves supporting the legalization case of MEAN where axis == [1, 2], keep_dims == True and input dtype == 'uint8'.

### CODE REVIEW

**Comment by ilyag-grovety** [comment]
cc @neildhickey @lhutton1 @ekalda @leandron @Aleksei-grovety

**Comment by ilyag-grovety** [comment]
@tvm-bot rerun

**Comment by ilyag-grovety** [comment]
@tvm-bot rerun

File: tests/python/contrib/test_ethosu/test_codegen.py
[the remark this answers is not in the document]
Comment (lhutton1) [reply tests/python/contrib/test_ethosu/test_codegen.py:1]: I believe the fix is part of Vela 3.7.0, so we would need to wait for that update to re-enable the tests

**Review by lhutton1 — APPROVED** [review]
Thanks @ilyag-grovety! I think @Aleksei-grovety's comment still needs addressing, otherwise LGTM

File: src/relay/op/contrib/ethosu/pooling.cc [hunk 1]
```diff
@@ -61,15 +61,27 @@ bool EthosuPoolingRel(const Array<Type>& types, int num_inputs, const Attrs& att
...
                 param->pooling_type);
 
+  DataType ofm_dtype = DataTypeFromString(param->ofm_dtype);
+
+  std::initializer_list<DataType> max_avg_pooling_ofm_dtypes = {DataType::Int(8), DataType::UInt(8),
+                                                                DataType::Int(16)};
+  if (param->pooling_type == "AVG" || param->pooling_type == "MAX") {
+    CheckDataType(reporter, ofm_dtype, max_avg_pooling_ofm_dtypes, operator_name, "ofm",
+                  param->pooling_type);
+    CheckDataTypeMatch(reporter, ofm_dtype, ifm->dtype, operator_name, "ifm", "ofm",
```
[the remark this answers is not in the document]
Comment (ilyag-grovety) [reply src/relay/op/contrib/ethosu/pooling.cc:78]: Proposed implementation keeps the status quo. There is an implicit restriction that ifm and ofm dtypes match. In this PR we just make it explicit. I suggest addressing this in further PRs by allowing ifm and ofm dtypes not match in specific cases of pooling operations.

[the remark this answers is not in the document]
Comment (lhutton1) [reply src/relay/op/contrib/ethosu/pooling.cc:78]: Happy to pull into a separate PR, thanks :)

**Pull request merged**